vllm/v1/core/sched/scheduler.py

```diff
@@ -274,6 +274,7 @@ def schedule(self) -> SchedulerOutput:
                         token_budget += num_scheduled_tokens[preempted_req.request_id]
                         req_to_new_blocks.pop(preempted_req.request_id)
                         num_scheduled_tokens.pop(preempted_req.request_id)
+                        req_index -= 1
```

Comment:
Thanks for catching this @CLFutureX and sorry for not getting to it sooner!

I think whether we decrement `req_index` here should depend on whether the index of the preempted request is less than `req_index`?